charts/zitadel/templates/setupjob.yaml

```
@@ -79,8 +79,25 @@ spec:
             mountPath: /config
           - name: chowned-secrets
             mountPath: /.secrets
+          {{- if and (not (empty .Values.zitadel.configmapConfig.FirstInstance)) (and (not (empty .Values.zitadel.configmapConfig.FirstInstance.Org)) (not (empty .Values.zitadel.configmapConfig.FirstInstance.Org.Machine))) }}
+          - name: machinekey
+            mountPath: {{ dir .Values.zitadel.configmapConfig.FirstInstance.MachineKeyPath }}
+          {{- end}}
           resources:
             {{- toYaml .Values.setupJob.resources | nindent 14 }}
+        {{- if and (not (empty .Values.zitadel.configmapConfig.FirstInstance)) (and (not (empty .Values.zitadel.configmapConfig.FirstInstance.Org)) (not (empty .Values.zitadel.configmapConfig.FirstInstance.Org.Machine))) }}
+        - name: "{{ .Chart.Name}}-machinekey"
+          image: alpine/k8s:1.25.2
+          command: [ "sh","-c","until [ ! -z $(kubectl get po ${POD_NAME} -o jsonpath=\"{.status.containerStatuses[?(@.name=='{{ .Chart.Name }}-setup')].state.terminated}\") ]; echo 'waiting for {{ .Chart.Name }}-setup container to terminate'; do sleep 5; done && echo '{{ .Chart.Name }}-setup container terminated' && if [ -f {{ .Values.zitadel.configmapConfig.FirstInstance.MachineKeyPath }} ]; then kubectl -n {{ .Release.Namespace }} create secret generic {{ .Values.zitadel.configmapConfig.FirstInstance.Org.Machine.Machine.Username }} --from-file={{ .Values.zitadel.configmapConfig.FirstInstance.Org.Machine.Machine.Username }}.json={{ .Values.zitadel.configmapConfig.FirstInstance.MachineKeyPath }}; fi;" ]
```

This also works if the setup doesn't create a file, you said?

```suggestion
          command: [ "sh","-c","until [ ! -z $(kubectl -n {{ .Release.Namespace }} get po ${POD_NAME} -o jsonpath=\"{.status.containerStatuses[?(@.name=='{{ .Chart.Name }}-setup')].state.terminated}\") ]; echo 'waiting for {{ .Chart.Name }}-setup container to terminate'; do sleep 5; done && echo '{{ .Chart.Name }}-setup container terminated' && if [ -f {{ .Values.zitadel.configmapConfig.FirstInstance.MachineKeyPath }} ]; then kubectl -n {{ .Release.Namespace }} create secret generic {{ .Values.zitadel.configmapConfig.FirstInstance.Org.Machine.Machine.Username }} --from-file={{ .Values.zitadel.configmapConfig.FirstInstance.Org.Machine.Machine.Username }}.json={{ .Values.zitadel.configmapConfig.FirstInstance.MachineKeyPath }}; fi;" ]
```